include/lsst/afw/geom/Transform.h

```diff
@@ -189,6 +189,28 @@ class Transform {
      */
     Eigen::MatrixXd getJacobian(FromPoint const &x) const;
 
+    /**
+     * Concatenate two Transforms.
+     *
+     * @tparam LastEndpoint the starting Endpoint of the Transform that feeds
+     *                      into this one
+     * @param before the Transform to apply before this one
+     * @returns a Transform that first applies `before` to its input, and then
+     *          this Transform to the result. Its inverse shall first apply the
+     *          inverse of this Transform, then the inverse of `before`.
```

Comment:
I'm finding this hard to follow. The template parameter `LastEndpoint` name seems backwards to me, which makes the rest of it harder for me to follow.

See if something like the following works for you:

I strongly suggest you name the argument `first`, just like ast::Mapping::of. Commonality is a huge help here. (I realize I would naturally prefer `first` since that's what I chose for astshim. If you are very keen for `before` we could consider changing it in both places. But I find `first` clear and concise.)

Then document could look something like this:

Concatenate two transforms to produce a new transform C(x) = this(first(x))

The resulting transform has the "from" endpoint of `first` (FromFirstEndpoint), and the "to" endpoint of this transform (ToEndpoint). The other two endpoints, where the transforms are joined, are required to be identical: thus the "to" transform of `first` must be the same as "from" endpoint of this transform: `FromEndpoint`. The compiler assures that the classes are the same, but a runtime check is required to make sure the number of axes match.

    template<class FirstFromEndpoint>
    Template<class FirstFromEndpoint, ToEndpoint> of(Template<FirstFromEndpoint, FromEndpoint> const & first)